samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImplGraph.java

```diff
@@ -323,70 +333,81 @@ public void init(Config config, TaskContext context) {
     };
   }
 
-  private boolean hasIntermediateStreams(OperatorSpecGraph specGraph) {
-    return !Collections.disjoint(specGraph.getInputOperators().keySet(), specGraph.getOutputStreams().keySet());
-  }
+  protected static class OperatorImplGraphUtil {
```

Comment:
Could you clarify the benefit of extracting this out?
One benefit I could see is if you wanted to mock this, but I don't see any mocking of this class.
It just looks a bit odd that you have a "util" class which you are using as an instance variable.

Reply:
These previously static util methods now need access to the `config` object to get the SystemStream for the streamId. I can either pass the StreamConfig through all of these explicitly and update all their tests, or make this a util class that can be tested separately. I chose the latter to minimize test refactors, but can do the former if that's preferable.

Reply:
If we were writing this from scratch, I think my preference would be to pass the StreamConfig through all of the static methods, so my vote would be to do that. Not a big deal if someone feels strongly about doing it the other way though.

Reply:
Sure, will pass the StreamConfig to the methods and update the tests.